### ISSUE TITLE
Searching an artist name shows all their songs already added to the library

### ISSUE THREAD

**Issue by darumanzane** [z-huang/music#1547]
### Checklist

- [X] I've checked that there is no other issue about this feature request.
- [X] This issue contains only one feature request.
- [X] The title of this issue accurately describes the feature request.

### Feature description

![IMG-20240903-WA0018](https://github.com/user-attachments/assets/c2204794-55de-4684-9473-b6815b6ce31d)

Searching should show entire artists discography when the prompt corresponds with their name, just like in the latest version of the app when you search inside a playlist:

![IMG-20240903-WA0017](https://github.com/user-attachments/assets/129c7f4b-dd10-4d83-9619-c31b8c6bc98d)

### Why do you want this feature?

![IMG-20240903-WA0019](https://github.com/user-attachments/assets/8d23bf9e-ea8d-4562-9d78-945401e6cbe2)

For example, if I'm looking for Doja Cat's songs, I'm going to try and see which of her songs I have saved in my library, but the way is implemented at the moment only shows songs that have Doja Cat in the tittle (like "ft. Doja Cat"). This makes harder to find things, since the discography of the artist won't have their name in the song most of the time.

### Additional information

![IMG-20240903-WA0016](https://github.com/user-attachments/assets/16e57e77-1a04-4554-bd3a-77da5e3c51fa)

Additionally, this strict way of showing results will make finding things difficult if you put a space at the end of the prompt, which is an automatic feature some devices have when you use the predictive text input. So "Cat " will get a different result that inputting "Cat", rendering some prompts unusable.

In other words, I'm not asking to see all the discography in general, but the specific songs I already added to my library. Plus, if you're not connected to the internet, you can't access an artist profile. But even offline you can search the things you have in your library.
